### ISSUE TITLE
Validator fails to read manifest.json

### ISSUE THREAD

**Issue by erosman** [mozilla/addons#888]
Today, validator fails to read `manifest.json`.

Tested on: https://addons.mozilla.org/en-US/developers/addon/validate

> Your add-on failed validation with 2 errors.
> 
>     Legacy extensions are no longer supported in Firefox.
>     This tool only works with legacy add-ons. See https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions for more information about WebExtension APIs.

Tested on: https://addons.mozilla.org/en-US/developers/addon/****/versions/submit/

> Your add-on failed validation with 3 errors.
> 
>     Could not parse the manifest file.
>     Legacy extensions are no longer supported in Firefox.
>     This tool only works with legacy add-ons. See https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions for more information about WebExtension APIs.

To **confirm** the origin of the error, I tested with a zip that was already uploaded to AMO (previous version of add-ons, I keep an archive of all) and the one that previously had passed, failed with the same error.

**Comment by kewisch** [comment]
I've had a second report of this via amo-admins email as well.

**Comment by wagnerand** [comment]
This is reproducible using a `manifest.json` using UTF-8 with BOM. ([Minimal test file](https://github.com/mozilla/addons/files/2803099/manifest.json.zip))


Stacktrace:
```
Uncaught exception:
  File \"/code/src/olympia/files/utils.py\", line 956, in parse_xpi
    xpi = get_file(xpi)
  File \"/code/src/olympia/files/utils.py\", line 155, in parse
    zip_file, certinfo=certificate_info).parse(minimal=minimal)
  File \"/code/src/olympia/files/utils.py\", line 387, in __init__
    self.data = decode_json(json_string)
  File \"/code/src/olympia/amo/utils.py\", line 173, in decode_json
    return json.loads(unicodehelper.decode(json_string))
  File \"/usr/local/lib/python2.7/json/__init__.py\", line 339, in loads
    return _default_decoder.decode(s)
  File \"/usr/local/lib/python2.7/json/decoder.py\", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File \"/usr/local/lib/python2.7/json/decoder.py\", line 382, in raw_decode
    raise ValueError(\"No JSON object could be decoded\")
<type 'exceptions.ValueError'>
ValueError('No JSON object could be decoded',)
```

bisecting says:
```d9634c91f2bfb6cddc3449370c8312a49f618261 is the first bad commit```

I think I nailed it down to:
https://github.com/mozilla/addons-server/commit/d9634c91f2bfb6cddc3449370c8312a49f618261#diff-910d9d00817510d6161315516feecfcbR367

`ipdb` test:
```
ipdb> zip_file.read('manifest.json')
'\xef\xbb\xbf{\n    "manifest_version": 2,\n    "version": "1.0",\n    "description": "A test extension from webext-generator.",\n    "name": "weafow-dotar-woonken"\n}'
ipdb> force_text(zip_file.read('manifest.json'))
u'\ufeff{\n    "manifest_version": 2,\n    "version": "1.0",\n    "description": "A test extension from webext-generator.",\n    "name": "weafow-dotar-woonken"\n}'
```

**Comment by diox** [comment]
I changed that again in https://github.com/mozilla/addons-server/pull/10525/files#diff-910d9d00817510d6161315516feecfcbL364

Will figure out what's going on. As @wagnerand suggests, since this happens only when `manifest.json` is encoded with a UTF-8 BOM, the workaround is to save the manifest without the BOM in any modern editor.

**Comment by erosman** [comment]
Note: The issue did NOT exist **10 days ago (Jan 18, 2019)** (and before that) when above v1.3 was uploaded to AMO. The issue has happened sometime during that last 10 days.

https://addons.mozilla.org/en-US/firefox/addon/firemonkey/

**Comment by wagnerand** [comment]
> Note: The issue did NOT exist 10 days ago (Jan 18, 2019) (and before that) when above v1.3 was uploaded to AMO. The issue has happened sometime during that last 10 days.

Yes, that commit was included in the push to production on Jan 23.

**Comment by diox** [comment]
https://github.com/mozilla/addons-server/pull/10525/files#diff-910d9d00817510d6161315516feecfcbL364 does fix that locally for me.

**Comment by wagnerand** [comment]
@diox so, can we close this then?

**Comment by wagnerand** [comment]
Next week or this week?

**Comment by diox** [comment]
Sorry, this week.

**Comment by AlexandraMoga** [comment]
Verified fixed on AMO -dev with FF64, Win10x64

Extensions with a `manifest.json` encoded with UTF-8-BOM are passing validation.

Example attached:
[UTF-8 BOM.zip](https://github.com/mozilla/addons/files/2808205/UTF-8.BOM.zip)
